crates/vm/src/system/memory/online.rs

```diff
@@ -143,35 +149,21 @@ impl GuestMemory {
 //     }
 // }
 
-// TO BE DELETED
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub enum MemoryLogEntry<T> {
-    Read {
-        address_space: u32,
-        pointer: u32,
-        len: usize,
-    },
-    Write {
-        address_space: u32,
-        pointer: u32,
-        data: Vec<T>,
-    },
-    IncrementTimestampBy(u32),
-}
-
-// perf[jpw]: since we restrict `timestamp < 2^29`, we could pack `timestamp, log2(block_size)`
-// into a single u32 to save half the memory, since `block_size` is a power of 2 and its log2
-// is less than 2^3.
 #[repr(C)]
 #[derive(Clone, Copy, PartialEq, Eq, Debug, derive_new::new)]
 pub struct AccessMetadata {
```

Comment:
Is TracingMemory still holding essentially a Vec of these via LinearMemory? If so, ideally we can make the size smaller: right now for touched_blocks iterating through the `meta` can be slow, also it is a good amount of memory. Alternatively, we could switch back to a slower paged vec approach to `meta`